plugins/woocommerce/src/Internal/ProductAttributesLookup/CLIRunner.php

```diff
@@ -407,20 +431,28 @@ private function regenerate_core( array $args = array(), array $assoc_args = arr
 
 		$this->data_regenerator->cancel_regeneration_scheduled_action();
 
+		$use_db_optimization = ! array_key_exists( 'disable-db-optimization', $assoc_args );
+		$this->check_can_use_db_optimization( $use_db_optimization );
 		$progress = WP_CLI\Utils\make_progress_bar( '', $products_count );
 		$this->log( "Regenerating %W{$table_name}%n..." );
-		if ( $processed_count > 0 ) {
-			$progress->tick( $processed_count );
-		}
+		$progress->tick( $processed_count );
 
-		while ( $this->data_regenerator->do_regeneration_step( $batch_size ) ) {
+		$regeneration_step_failed = false;
+		while ( $this->data_regenerator->do_regeneration_step( $batch_size, $use_db_optimization ) ) {
 			$progress->tick( $batch_size );
+			$regeneration_step_failed |= $this->data_regenerator->get_last_regeneration_step_feiled();
```

Comment:
> I'm wondering if there's a specific reason to use it rather than just doing

It makes sense if you look at the complete loop:

```php
$regeneration_step_failed = false;
while ( $this->data_regenerator->do_regeneration_step( $batch_size, $use_db_optimization ) ) {
	$progress->tick( $batch_size );
	$regeneration_step_failed |= $this->data_regenerator->get_last_regeneration_step_failed();
}
```

I want the variable to be true if _at least one_ of the steps fails, so I'm ORing all the values together; thus the final value of the variable will be true if at least one of the intermediate values was true.

That said: indeed, the `|` operator (or the variant I was using, `|=`) is the _bitwise_ OR operator and while it works with booleans, it's more appropriate to use `||` instead, which is the _logical_ OR operator. There's no `||=` operator in PHP unfortunately, so I've changed it to use `||` repeating the variable name in the left and right sides of the expression.